### PR TITLE
nodes metadata JSON ouput

### PR DESCRIPTION
**Pull request by crazy-max**
At the moment `endpoint`, `status`, `flags` and `platforms` outputs are only return for the very first note being registered for the builder.

This PR adds a new output `nodes` that returns metadata JSON of all registered nodes and also adds new fields available since Buildx 0.9:

```json
[
  {
    "name": "builder-06133552-7a7d-4d4f-a011-fccc08c3ae500",
    "endpoint": "unix:///var/run/docker.sock",
    "driver-opts": [
      "image=moby/buildkit:master",
      "network=host"
    ],
    "status": "running",
    "buildkitd-flags": "--allow-insecure-entitlement security.insecure --allow-insecure-entitlement network.host",
    "buildkit": "81b6ff2",
    "platforms": "linux/amd64,linux/amd64/v2,linux/amd64/v3,linux/386"
  }
]
```

This change is primarily for future support of appending nodes to a builder, so the output is relevant.

`endpoint`, `status`, `flags` and `platforms` outputs are deprecated and will be removed in next major release.

### CODE REVIEW

**Review by tonistiigi — COMMENTED** [review]
I could maybe see `status` and `platforms` being a little bit useful in some cases. If the goal is to handle the cases with multiple nodes(not very common in CI) then maybe we could just combine the values for these fields?

File: src/main.ts [hunk 1]
```diff
@@ -105,21 +105,26 @@ async function run(): Promise<void> {
...
       core.startGroup(`BuildKit version`);
-      core.info(await buildx.getBuildKitVersion(`buildx_buildkit_${builder.node_name}`));
+      for (const node of builder.nodes) {
+        const bkvers = await buildx.getBuildKitVersion(`buildx_buildkit_${node.name}`);
+        core.info(`${node.name}: ${bkvers}`);
+      }
       core.endGroup();
     }
-    if (core.isDebug() || builder.node_flags?.includes('--debug')) {
+    if (core.isDebug() || firstNode['buildkitd-flags']?.includes('--debug')) {
```
Comment (jedevc) [inline src/main.ts:127]: Not super familiar with the logic, but would we maybe want this if any of those nodes have the `--debug` flag, not just the first?

File: README.md [hunk 2]
```diff
@@ -244,14 +244,11 @@ Following inputs can be used as `step.with` keys
...
-| `driver`      | String  | Builder driver |
-| `endpoint`    | String  | Builder node endpoint |
-| `status`      | String  | Builder node status |
-| `flags`       | String  | Builder node flags (if applicable) |
-| `platforms`   | String  | Builder node platforms available (comma separated) |
+| Name        | Type   | Description            |
+|-------------|--------|------------------------|
+| `name`      | String | Builder name           |
+| `driver`    | String | Builder driver         |
+| `nodes`     | JSON   | Builder nodes metadata |
```
Comment (jedevc) [inline README.md:251]: We should describe the JSON schema of the `nodes` metadata, maybe in a table below?

**Review by jedevc — COMMENTED** [review]
> I could maybe see `status` and `platforms` being a little bit useful in some cases. If the goal is to handle the cases with multiple nodes(not very common in CI) then maybe we could just combine the values for these fields?

Agreed, especially makes sense for `platforms`, I like the idea of merging each node's together for this if we can.

**Comment by crazy-max** [comment]
> I could maybe see `status` and `platforms` being a little bit useful in some cases. If the goal is to handle the cases with multiple nodes(not very common in CI) then maybe we could just combine the values for these fields?

> Agreed, especially makes sense for `platforms`, I like the idea of merging each node's together for this if we can.

This would not give the expected result as we can define preferred platforms even if those are not supported on specified nodes:

```
  {
    "nodes": [
      {
        "name": "builder-13dbff8e-3be6-4024-9aa5-2c75782f1cf10",
        "endpoint": "docker-container://buildx_buildkit_builder-ba6a88cb-fe10-4412-876e-66458fe474180",
        "status": "running",
        "platforms": "linux/amd64,linux/amd64/v2,linux/amd64/v3,linux/amd64/v4,linux/arm64,linux/riscv64,linux/ppc64le,linux/s390x,linux/386,linux/mips64le,linux/mips64,linux/arm/v7,linux/arm/v6"
      },
      {
        "name": "aws_graviton2",
        "endpoint": "tcp://***:1234",
        "driver-opts": [
          "cacert=/home/runner/.docker/buildx/creds/builder-13dbff8e-3be6-4024-9aa5-2c75782f1cf1/cacert_***-1234.pem",
          "cert=/home/runner/.docker/buildx/creds/builder-13dbff8e-3be6-4024-9aa5-2c75782f1cf1/cert_***-1234.pem",
          "key=/home/runner/.docker/buildx/creds/builder-13dbff8e-3be6-4024-9aa5-2c75782f1cf1/key_***-1234.pem"
        ],
        "status": "running",
        "platforms": "darwin/arm64*,linux/arm64*,linux/arm/v5*,linux/arm/v6*,linux/arm/v7*,windows/arm64*"
      },
      {
        "name": "linuxone_s390x",
        "endpoint": "tcp://***:1234",
        "driver-opts": [
          "cacert=/home/runner/.docker/buildx/creds/builder-13dbff8e-3be6-4024-9aa5-2c75782f1cf1/cacert_***-1234.pem",
          "cert=/home/runner/.docker/buildx/creds/builder-13dbff8e-3be6-4024-9aa5-2c75782f1cf1/cert_***-1234.pem",
          "key=/home/runner/.docker/buildx/creds/builder-13dbff8e-3be6-4024-9aa5-2c75782f1cf1/key_***-1234.pem"
        ],
        "status": "running",
        "platforms": "linux/s390x*"
      }
    ],
    "name": "builder-13dbff8e-3be6-4024-9aa5-2c75782f1cf1",
    "driver": "remote"
  }
```

For example here `aws_graviton2` supported platforms are:

```
time="2022-09-19T21:12:20Z" level=info msg="found worker \"klrqif3uuxy84t69eo3eef89n\", labels=map[org.mobyproject.buildkit.worker.executor:oci org.mobyproject.buildkit.worker.hostname:7c099955516d org.mobyproject.buildkit.worker.network:host org.mobyproject.buildkit.worker.oci.process-mode:sandbox org.mobyproject.buildkit.worker.snapshotter:overlayfs], platforms=[linux/arm64 linux/arm/v7 linux/arm/v6]"
```

But I define the following ones as preferred:

```yaml
      -
        name: Set up Docker Buildx
        uses: crazy-max/docker-setup-buildx-action@append
        with:
          driver: remote
          endpoint: docker-container://buildx_buildkit_${{ steps.builder.outputs.name }}0
          append: |
            - name: aws_graviton2
              endpoint: tcp://${{ secrets.AWS_ARM64_HOST }}:1234
              platforms: darwin/arm64,linux/arm64,linux/arm/v5,linux/arm/v6,linux/arm/v7,windows/arm64
            - name: linuxone_s390x
              endpoint: tcp://${{ secrets.LINUXONE_S390X_HOST }}:1234
              platforms: linux/s390x
```

See #165 about `append` input

**Comment by crazy-max** [comment]
As discussed keep `platforms` output but return preferred ones if set or all available.

**Review by jedevc — APPROVED** [review]

**Pull request merged**